### PR TITLE
[Test] Consolidate `doSelectTarget()` into `doAttack()`

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
There are no user-facing changes.

## Why am I doing these changes?
Suggestion by snooz on Discord.

## What did change?
`doAttack()` now takes an optional `targetIndex` parameter and calls `doSelectTarget()` unless `targetIndex` is `null`.

## How to test the changes?
`npm run test`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- ~[ ] Have I considered writing automated tests for the issue?~
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

File: src/test/utils/gameManager.ts [hunk 1]
```diff
@@ -181,35 +181,34 @@ export default class GameManager {
...
    * usually called after {@linkcode doAttack} in a double battle.
    * @param {BattlerIndex} targetIndex the index of the attack target
    */
-  doSelectTarget(targetIndex: BattlerIndex) {
+  doSelectTarget(targetIndex: BattlerIndex, movePosition: integer) {
     this.onNextPrompt("SelectTargetPhase", Mode.TARGET_SELECT, () => {
       const handler = this.scene.ui.getHandler() as TargetSelectUiHandler;
-      handler.setCursor(targetIndex);
+      const move = (this.scene.getCurrentPhase() as SelectTargetPhase).getPokemon().getMoveset()[movePosition]!.getMove(); // TODO: is the bang correct?
+      if (!move.isMultiTarget()) {
```
Comment (snoozbuster) [inline src/test/utils/gameManager.ts:207]: I wonder if it's worth throwing an error if the move is multi target and an explicit target is provided. Maybe save someone the trouble of asking "why is my target selection not working" when they don't realize their test move is multitarget

Comment (DayKev) [reply src/test/utils/gameManager.ts:207]: Good idea.

File: src/test/utils/gameManager.ts [hunk 2]
```diff
@@ -181,35 +181,34 @@ export default class GameManager {
...
+    }
   }
 
   /**
    * Emulate a player's target selection after an attack is chosen,
    * usually called after {@linkcode doAttack} in a double battle.
    * @param {BattlerIndex} targetIndex the index of the attack target
    */
-  doSelectTarget(targetIndex: BattlerIndex) {
+  doSelectTarget(targetIndex: BattlerIndex, movePosition: integer) {
```
Comment (torranx) [inline src/test/utils/gameManager.ts:203]: update docs pls

File: src/test/utils/gameManager.ts [hunk 3]
```diff
@@ -181,35 +181,34 @@ export default class GameManager {
    * Emulate a player attack
    * @param movePosition the index of the move in the pokemon's moveset array
    */
-  doAttack(movePosition: integer) {
+  doAttack(movePosition: integer, targetIndex?: BattlerIndex | null) {
```
Comment (torranx) [inline src/test/utils/gameManager.ts:184]: update docs pls

**Review by torranx — APPROVED** [review]

**Review by snoozbuster — APPROVED** [review]

**Comment by DayKev** [comment]
Waiting for #3567

**Pull request closed without merging**